eng/scripts/validate-changes-on-allowed-path.ps1

```diff
@@ -0,0 +1,82 @@
+<#
+.SYNOPSIS
+
+.DESCRIPTION
+The script is to validate if the diff changes between base and target are within the scope of allowlist.
+
+.PARAMETER BaseBranch
+The base branch where we want to merge changes from. E.g. main branch in docs repo
+
+.PARAMETER TargetBranch
+The target branch where we want to merge changes to. E.g. live branch in docs repo
+
+.PARAMETER AllowlistPath
+The file path of allowlist.
+#>
+param(
+  [Parameter(mandatory=$true)]
+  [string] $BaseBranch,
+  [Parameter(mandatory=$true)]
+  [string] $TargetBranch,
+  [Parameter(mandatory=$true)]
+  [string] $AllowlistPath
+)
+
+Set-StrictMode -Version 3
+. (Join-Path $PSScriptRoot ../common/scripts/common.ps1)
+
+# Fetch all branches, e.g. main, live
+Write-Host "git -c user.name=`"azure-sdk`" -c user.email=`"azuresdk@microsoft.com`" fetch origin"
+git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" fetch origin
+
+Write-Host "Check existence on target branch"
+# Checking whether the target branch exists. Suppress the error message and use the exit code for error handling.
+git rev-parse --verify $TargetBranch 2>$null
+if ($LASTEXITCODE -ne 0) {
+  Write-Host "Creating target branch $TargetBranch..."
+  git checkout -b $TargetBranch -t origin/$TargetBranch
```

Comment:
Why do we need to checkout this branch given we are checking out the base branch below?

Reply:
We have to have the live branch in local to do the comparison.

Reply:
I changed to compare with the origin/$targetbranch